### ISSUE TITLE
Support for MacOS

### ISSUE THREAD

**Issue by jposterloh** [documents4j/documents4j#126]
Hi!

We have some guys who work with MacOS. Any idea on supporting that? We use Excel to PDF conversion. 

I see that you execute some VB-Scripts, could this also be done with AppleScript, or some bash scripting? I would be able to help, if it is possible and if you tell me what would be needed. 

Best regards
Jan-Patrick

**Comment by raphw** [comment]
Absolutely, the logic for VBS is rather isolated, it should be straight forward to create an abstraction to also support MacOS. If you send me a PR, I will certainly consider it!

**Comment by jposterloh** [comment]
Hi, sorry for not responding for this long, but we had a quite stressful time here, and more important things. 

I've now figured out some AppleScripts to mimic the VBS scripts. 

I identified for Excel two areas where it has to be changed for MAC: 
- [documents4j-transformer-msoffice-excel] "com.documents4j.conversion.msoffice.MicrosoftExcelScript" has to use ".applescript" instead of ".vbs"
- [documents4j-transformer-msoffice-base]"com.documents4j.conversion.msoffice.AbstractMicrosoftOfficeBridge" in the doStartConversion. 

My questions: 
- Did I miss any code that had to be made platform specific?
- Do you have any preferred way on how to introduce the correct extension? i.e. just a check in the constructor for the file type, or something more sophisticated like a factory that delivers a system dependent version of the Script implementation?
- Is there already a functions in the utilities to check the OS version? Or dependencies that contain such functionality? If not, where do you want me to put it? in the msoffice-base or in some util module?

I'd like to start with the excel, but will then adapt the word and powerpoint scripts, too. 

I hope to get access to a Mac PC next week, so I will try to implement everything then. 

Bye

**Comment by raphw** [comment]
No, if you get the scripts to work, then it is no longer platform specific. Ideally, for compatibility, you would discover if you are running on Windows or on a Mac and unpack the corresponding scripts.

Looking forward to the contribution.

**Comment by jposterloh** [comment]
Hi, 

so the good news:
- I have a working script for Excel 2016 on OS X Big Sur creating PDF from Excel
- The script is executed correctly in documents4j

However, there is now one problem, and that is that the return value of the AppleScript is NOT returned as the Process's exitValue, but instead written to stdout. And this then results in the "MicrosoftOfficeTargetNameCorrector" to return "false" from the "conversionSuccessful", making the process failing. 

Any idea how to handle this, because this breaks a lot...

Best
Jan-Patrick

**Comment by raphw** [comment]
Wouldn't this be possible to achieve by some minor changes to `AbstractExternalConverter`? Here is a suggestion:

https://github.com/documents4j/documents4j/commit/e199e746c24e20e2d8f21f5b49ce1e4219d0901e

**Comment by paulohcasanova** [comment]
Hi guys.
 
I've tried to use the `1.1.12` version to convert a DOCX file to PDF however I get the following error :
`[Thread-3] INFO com.documents4j.conversion.msoffice.MicrosoftWordBridge - ERROR: Microsoft Word got an error: document doesn’t understand the “save as” message.`
`[Thread-2] INFO com.documents4j.conversion.msoffice.MicrosoftWordBridge - -6`
 
Is it possible that there is an issue based on the MS Word version on the Mac ?
 
Code used for conversion :
```
File inputWord = new File(docxFilePath);
File outputFile = new File(pdfFilePath);
InputStream docxInputStream = new FileInputStream(inputWord);
OutputStream outputStream = new FileOutputStream(outputFile);
IConverter converter = LocalConverter.builder().build();
converter.convert(docxInputStream).as(DocumentType.DOCX).to(outputStream).as(DocumentType.PDF).execute();
outputStream.close();
```
